core/src/main/java/jenkins/mvn/GlobalSettingsProvider.java

```diff
@@ -21,13 +26,45 @@
 public abstract class GlobalSettingsProvider extends AbstractDescribableImpl<GlobalSettingsProvider> implements ExtensionPoint {
 
     /**
+     * <p>
+     *     Configure maven launcher argument list with adequate settings path.
+     * </p>
+     * <p>Implementations should
+     * <ul>Be aware that this method might get called multiple times during a build.</ul>
+     * <ul>Implement this method. This class provides a default implementation throwing an {@link UnsupportedOperationException}
+     * so that implementations have time to adapt.</ul>
+     * </p>
+     *
+     * @param run       the build / run to provide the settings for
+     * @param workspace the workspace in which the build / run takes place
+     * @param listener the listener of this given build / run
+     * @return the filepath to the provided file. <code>null</code> if no settings will be provided.
+     */
+    @CheckForNull
+    public FilePath supplySettings(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnull TaskListener listener) throws IOException, InterruptedException {
+        throw new AbstractMethodError("Class " + getClass() + " must override the new supplySettings overload");
+    }
+
+    /**
      * configure maven launcher argument list with adequate settings path
      * 
      * @param build
-     *            the build to provide the settigns for
+     *            the build to provide the settings for
      * @return the filepath to the provided file. <code>null</code> if no settings will be provided.
+     * @deprecated use {@link #supplySettings(Run, FilePath, TaskListener)}
```

Comment:
Without the fix proposed above it would be a wrong guidance since the provider implementations may not support the legacy API